timml/besselaesnew/besselaesnew.py

```diff
@@ -0,0 +1,706 @@
+    # Radius of convergence
+    Rconv = 7.0
+
+    # lstype=2 means line-doublet
+    lstype = 2
+
+    zin, z1in, z2in, Lin, z, zplus1, zmin1 = prepare_z(x, y, z1, z2)
+
+    # Laplace line-doublet
+    qm = np.complex(0.0, 0.0)
+    if ilap == 1:
+        if order == 0:
+            wdis = -(1.0 / zmin1 - 1.0 / zplus1) / (
+                np.pi * np.complex(0.0, 1.0) * (z2in - z1in)
+            )
+        else:
+            wdis = float(order) * z ** (order - 1) * np.log(zmin1 / zplus1)
+            wdis = wdis + z ** order * (1.0 / zmin1 - 1.0 / zplus1)
+            qm = np.complex(0.0, 0.0)
+            if order > 1:  # To avoid a possible problem of 0 * 0^(-1)
+                for n in range(1, int(order / 2) + 1):
+                    qm = qm + float(order - 2 * n + 1) * z ** (order - 2 * n) / float(
+                        2 * n - 1
+                    )
+
+            wdis = -(wdis + 2.0 * qm) / (np.pi * np.complex(0.0, 1.0) * (z2in - z1in))
+
+        rv[0, 0] = np.real(wdis)
+        rv[1, 0] = -np.imag(wdis)
+
+    # N-1 or N leakage factors
+    for i in range(ilap, naq):
+        wdis = np.complex(0.0, 0.0)
+
+        # Check whether entire line-doublet is outside radius of convergence
+        # Outside if |z-zc|>L/2+7lab, and thus |Z|>1+7lab*2/L, or |zeta|>1/biglab+7 (zeta is called z here)
+        biglab = 2.0 * labda[i] / Lin
+        z = (2.0 * zin - (z1in + z2in)) / (z2in - z1in) / biglab
+
+        if abs(z) < (Rconv + 1.0 / biglab):
+            m1, m2, NLS = findm1m2(zin, z1in, z2in, Lin, labda[i], Rconv)
+            wdis1 = np.complex(0.0, 0.0)
+            if m1 > 0:
+                z1new = z1in + float(m1 - 1) / float(NLS) * (z2in - z1in)
+                z2new = z1in + float(m2) / float(NLS) * (z2in - z1in)
+                del0 = float(1 - m1 - m2 + NLS) / float(1 - m1 + m2)
+                ra = float(NLS) / float(1 + m2 - m1)
+                wdis1 = IntegralLapLineDipoleDis(zin, z1new, z2new, del0, ra, order)
+                wdis1 = -2.0 * wdis1 / (np.complex(0.0, 1.0) * (z2new - z1new))
+
+            pot = IntegralF(zin, z1in, z2in, Lin, labda[i], order, Rconv, lstype)
+            wdis2 = IntegralG(zin, z1in, z2in, Lin, labda[i], order, Rconv, lstype)
+            wdis3 = pot / (2.0 * np.complex(0.0, 1.0)) + wdis2 * np.imag(z)
+
+            wdis = wdis1 - 4.0 * wdis3 / (biglab ** 2 * (z2in - z1in))
+        rv[0, i] = np.real(wdis)
+        rv[1, i] = -1.0 * np.imag(wdis)
+    return rv
+
+
+@numba.njit
+def disbesldv(x, y, z1, z2, lab, order, ilap, naq):
+    qxqy = np.zeros((2 * (order + 1), naq))
+    # Check if endpoints need to be adjusted using the largest labda (the first one)
+    for n in range(0, order + 1):
+        rv = disbesldho(x, y, z1, z2, lab, n, ilap, naq)
+        qxqy[n, 0 : naq + 1] = rv[0, 0:naq]
+        qxqy[n + order + 1, 0 : naq + 1] = rv[1, 0 : naq + 1]
+    return qxqy
+
+
+@numba.njit
+def IntegralF(zin, z1in, z2in, Lin, labda, order, Rconv, lstype):
```

Comment:
Is there perhaps an alternative in the `scipy` family of packages that does this in C code already? The variable and function naming makes it somewhat hard to grasp what's exactly happening, so I can't offer a direct replacement, but there are various Bessel functions in `scipy.special`.